### ISSUE TITLE
XMonad.Prompt.RunOrRaise always opens a directory that has the same name as an executable.

### ISSUE THREAD

**Issue by yusi1** [xmonad/xmonad-contrib#616]
### Problem Description
Hello everyone again, I've been messing around with XMonad.Prompt.RunOrRaise and since it can open both directories and executables (in /usr/bin); I've been having an issue where a directory that has the same name as an executable causes the directory to _always_ be shown in a file manager, with the executable _never_ being executed; An example of this is in this scenario: I type `vmware` into the prompt, the `vmware` program has a directory in my home dir that has the same name as the executable in `/usr/bin`, (the prompt shows 2 matches: `vmware` and `vmware/`), if I select `vmware` thunar file manager launches and shows the `vmware` directory, if I select `vmware/` thunar file manager launches and shows the `vmware` directory aswell. Shouldn't `vmware` be the executable and `vmware/` be the dir?? It should not launch in both cases...

#### Example:
- Setup XMonad.Prompt.RunOrRaise...
- Create a directory in the users home directory that has the same name as an executable in `/usr/bin` (for example: "vmware")
- The prompt shows 2 matches: `vmware` and `vmware/` both do the same thing, but they shouldn't
- The directory is *always* opened, with the executable *never* being executed...

### Configuration File

Please include the smallest _full_ configuration file that reproduces
the problem you are experiencing:

```haskell
module Main (main) where

import XMonad
import XMonad.Prompt
import XMonad.Prompt.RunOrRaise

-- Basic XPConfig
myXPConfig :: XPConfig
myXPConfig = def { font = "xft:PragmataPro Mono Liga:Bold:size=18"
                                , bgColor = "black"  
                                , fgColor = "green"
                                , bgHLight = "green"
                                , fgHLight = "black"
                                , borderColor = "#262626"
                                , promptBorderWidth = 1
                                , position = Top    
                                , height = 24   
                                , searchPredicate = fuzzyMatch
                                , sorter = fuzzySort
                  }

-- Main ...
-- Shortcuts for RunOrRaisePrompt ...
```

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - I tested my configuration
    - [x] With `xmonad` version 0.15
    - [x] With `xmonad-contrib` version 0.16

**Comment by yusi1** [comment]
On the XMonad Matrix channel, Geekosaur suggested something similar to this, example: `vmware*` for a executable and `vmware/` for a file directory...

**Comment by yusi1** [comment]
Should directories be left to dirExec instead of being in the runAndRaise prompt?? I mean tbh I only use runOrRaise for launching executables anyway...

**Comment by slotThe** [comment]
I think it's still worth keeping this functionality around (though with some warnings in the docs) for people who need it _and_ whose setup can actually make use of it.

@yusi1 Can you confirm that the following works for you? http://ix.io/3B4P/diff  The fix for just directories is remarkably simple, for files we'd probably need to also suffix them with some string (I think snoc'ing `*` to all executables would introduce too much noise) and then take care to subtract that when necessary.  I'm not sure whether that's worth it tbh, especially since no-one so far has complained :)

**Comment by slotThe** [comment]
> How would I go about testing these changes, do I use the git version of XMonad-contrib?? I'm going to setup a VM to test these changes.. (Beginner btw)

Yes, you would have to build xmonad-contrib from the git version and apply this patch (actually, I'm going to open a PR so this is easier for you, you then only have to check out that branch so build).  There is a good introduction on how to set up a stack environment in [INSTALL.md](https://github.com/xmonad/xmonad/blob/master/INSTALL.md)
